posthog/models/event/query_event_list.py

```diff
@@ -85,19 +87,149 @@ def query_events_list(
+        context.collect_values = collected_hogql_values
+        clickhouse_sql = translate_hql(expr, context)
+        select_columns.append(clickhouse_sql)
+        if not context.is_aggregation:
+            group_by_columns.append(clickhouse_sql)
+
+    for expr in where or []:
+        context = ExprParserContext()
+        context.collect_values = collected_hogql_values
+        clickhouse_sql = translate_hql(expr, context)
+        if context.is_aggregation:
+            having_filters.append(clickhouse_sql)
+        else:
+            where_filters.append(clickhouse_sql)
+
+    if order_by:
+        for fragment in order_by:
+            order_direction = "ASC"
+            if fragment.startswith("-"):
+                order_direction = "DESC"
+                fragment = fragment[1:]
+            context = ExprParserContext()
+            context.collect_values = collected_hogql_values
+            order_by_list.append(translate_hql(fragment, context) + " " + order_direction)
     else:
-        return query_with_columns(
-            SELECT_EVENT_BY_TEAM_AND_CONDITIONS_SQL.format(conditions=conditions, limit=limit_sql, order=order),
-            {"team_id": team.pk, "limit": limit, **condition_params},
-        )
+        order_by_list.append(select_columns[0])
+
+    if select_columns == group_by_columns:
+        group_by_columns = []
+
+    results, types = sync_execute(
+        SELECT_EVENT_FIELDS_BY_TEAM_AND_CONDITIONS_FILTERS_PART.format(
+            columns=", ".join(select_columns),
+            conditions=conditions,
+            filters=prop_filters,
+            where="AND {}".format(" AND ".join(where_filters)) if where_filters else "",
+            group="GROUP BY {}".format(", ".join(group_by_columns)) if group_by_columns else "",
+            having="HAVING {}".format(" AND ".join(having_filters)) if having_filters else "",
+            order="ORDER BY {}".format(", ".join(order_by_list)),
+            limit=f"LIMIT {int(limit)}",
+        ),
+        {"team_id": team.pk, **condition_params, **prop_filter_params, **collected_hogql_values},
+        with_column_types=True,
+    )
+
+    # Convert star field from tuple to dict in each result
+    if "*" in select:
+        star = select.index("*")
+        for index, result in enumerate(results):
+            results[index] = list(result)
+            results[index][star] = convert_star_select_to_dict(result[star])
+
+    # Convert person field from tuple to dict in each result
+    if "person" in select:
+        person = select.index("person")
+        for index, result in enumerate(results):
+            results[index] = list(result)
+            results[index][person] = convert_person_select_to_dict(result[person])
+
+    return {
+        "results": results,
+        "columns": select,
+        "types": [type for _, type in types],
+    }
+
+
+def parse_order_by(order_by_param: Optional[str], select: Optional[List[str]]) -> List[str]:
+    if order_by_param:
```

Comment:
Is this a not null check or a "" check or both? IMO really worth being explicit with both this and `select` - it's a real pain to debug otherwise.

Reply:
Ugh, this function was some leftover that I appended to. Made it a [bit simpler](https://github.com/PostHog/posthog/pull/13203/commits/921b829bd33d4ace388fe4889db322ba600ca0e9).